sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala

```diff
@@ -114,4 +114,197 @@ object EstimationUtils {
     }
   }
 
+  /**
+   * Returns the number of the first bin/bucket into which a column values falls for a specified
+   * numeric equi-height histogram.
+   *
+   * @param value a literal value of a column
+   * @param histogram a numeric equi-height histogram
+   * @return the number of the first bin/bucket into which a column values falls.
+   */
+
+  def findFirstBucketForValue(value: Double, histogram: Histogram): Int = {
```

Comment:
Shall we unify all names to `bin`/`bins` in code and comments?

Reply:
We had bucket(s) and bin(s) used interchangeably.  To avoid confusion, I will unify them to use only bin/bins.